### PR TITLE
Add a faster implementation of TimeZoneSeries

### PR DESCRIPTION
**Pull request by mniip**

### CODE REVIEW

File: src/Holidays/EDay.hs [hunk 1]
```diff
@@ -7,7 +7,7 @@ import Data.ByteString.Char8
 import Data.Csv
 import Data.Time
 import Data.Time.Calendar.OrdinalDate
-import Data.Vector qualified as V
+import qualified Data.Vector as V
```
Comment (stevladimir) [inline src/Holidays/EDay.hs:10]: I think the consensus is to migrate gradually to ImportQualifiedPost.
Is it going to be used in Reflex frontends?

**Review by aleator — APPROVED** [review]

File: src/Data/Time/FastSeries.hs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+--
+-- The handling of these corner cases matches exactly to that of
+-- 'Series.localTimeToUTC''
+--
+-- This means the equation
+-- @lt == utcToLocalTime' tzsf (localTimeToUTC' tzsf lt)@ does not hold for @lt@
+-- that were skipped. And the equation
+-- @t == localTimeToUTC' tzsf (utcToLocalTime' tzsf t)@ does not hold for @t@
+-- that come just before switching awya from summer time. For other arguments
+-- these equations are assumed to hold.
```
Comment (stevladimir) [inline src/Data/Time/FastSeries.hs:82]: Is it worth to have tests for this?

File: src/Data/Time/FastSeries.hs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+--
+-- At the moment of switching to summer time, there is a range of 'LocalTime's
+-- that are skipped. For these, 'Series.isValidLocalTime' returns 'False'. This
+-- function converts them to 'UTCTime' as if the switch didn't happen.
+--
+-- Around the moment of switching away from summer time, there is a range of
+-- 'LocalTime's that happen twice: first before the switch, then once again
+-- after the switch. For these, 'Series.isRedundantLocalTime' returns 'True'.
+-- In such ambiguous cases, this function always picks the occurrence after the
+-- switch.
```
Comment (stevladimir) [inline src/Data/Time/FastSeries.hs:72]: Did you reach someone on business side to learn how it should actually work from their point of view?
Keeping existing behavior is OK, but if we can do it right, why not?

**Review by stevladimir — APPROVED** [review]

**Review by Yuras — APPROVED** [review]

File: src/Holidays/EDay.hs (hunk 1 above)
Comment (mniip) [reply src/Holidays/EDay.hs:10]: This package is used in GHCJS code, so I made it compile with 8.6

**Review by Yuras — APPROVED** [review]

File: src/Data/Time/FastSeries.hs (hunk 3 above)
Comment (aleator) [reply src/Data/Time/FastSeries.hs:72]: Iirc I asked once and no one had really a clue. Ask Maxim / Elena?

**Pull request merged**